src/app/components/applications/SSV/SetOperatorFee/SetOperatorFee.tsx

```diff
@@ -17,22 +17,31 @@ import OperatorStore from '~app/common/stores/applications/SsvWeb/Operator.store
 
 type UserInput = string;
 
+const INITIAL_ERROR_STATE = { shouldDisplay: false, errorMessage: '' };
+
 const SetOperatorFee = () => {
   const stores = useStores();
   const classes = useStyles();
   const navigate = useNavigate();
   const operatorStore: OperatorStore = stores.Operator;
+  const [error, setError] = useState(INITIAL_ERROR_STATE);
   const [zeroError, setZeroError] = useState(false);
-  const [userInput, setUserInput] = useState<UserInput | null>(null);
-  const [registerButtonEnabled, setRegisterButtonEnabled] = useState(true);
-  const [error, setError] = useState({ shouldDisplay: false, errorMessage: '' });
+  const [userInput, setUserInput] = useState<UserInput>('');
+  const [registerButtonDisabled, setRegisterButtonDisabled] = useState(true);
 
   useEffect(() => {
-    const isRegisterButtonEnabled = typeof userInput === 'object' || error.shouldDisplay;
-    setRegisterButtonEnabled(!isRegisterButtonEnabled);
-    return () => {
-      setRegisterButtonEnabled(false);
-    };
+   if ( userInput === '') {
+     setZeroError(false);
+     setError(INITIAL_ERROR_STATE);
+     setRegisterButtonDisabled(true);
+     return;
+    }
+    validateFeeInput(userInput, setError);
+    setUserInput(removeLeadingZeros(userInput));
+    if (userInput === '0') setZeroError(true);
+    else setZeroError(false);
```

Comment:
Please make this if else using `{}` brackets

Reply:
done